iati/resources.py

```diff
@@ -372,11 +464,52 @@ def resource_filesystem_path(path):
     Returns:
         str: A reference to the specified file that works however the package is distributed.
 
+    Raises:
+        TypeError: If the given path is of a type that cannot be a filepath.
+        ValueError: If the given path is a string that cannot be a useful component of a filepath.
+
     Note:
         Does not check to see that the specified file exists.
 
-    Warning:
-        When other functions in this module are reviewed, this will be too.
-
     """
+    try:
+        _ensure_portable_filepath(path)
+    except ValueError:
+        if path != '':
+            raise
+
     return pkg_resources.resource_filename(PACKAGE, path)
+
+
+def _ensure_portable_filepath(maybe_filepath):
+    """Determine whether a string could be a portable filepath.
+
+    A portable filepath is one that meets the POSIX fully portable filename character restrictions:
+
+    * Consists only of: A-Z a-z 0-9 . _ -
+    * Does not have a leading hyphen.
+
+    Args:
+        maybe_filepath (str): A string that might be considered a valid filepath.
+
+    Raises:
+        TypeError: If the input value is of a type that cannot be a filepath.
+        ValueError: If the input value is a string that cannot be a useful component of a filepath.
+
+    Todo:
+        Consider utilising the Python3.6 concept of path-like objects.
+
+    """
+    if not isinstance(maybe_filepath, str):
+        raise TypeError('A filesystem path must be a string. The provided value is a {0}.'.format(type(maybe_filepath)))
+
+    path_components = maybe_filepath.split(os.path.sep)
+
+    # allow there to be a trailing folder separator on the input maybe_filepath
+    if len(path_components) > 1 and path_components[-1] == '':
+        path_components.pop()
+
+    permitted_component_regex = re.compile('^[_.A-Za-z0-9][-_.A-Za-z0-9]*$')
+    for component in path_components:
+        if re.match(permitted_component_regex, component) is None:
+            raise ValueError('Each component in a permitted filepath must only include the following characters: A-Z a-z 0-9 . _ - (Problem component: {0} Actual path: {1})'.format(component, maybe_filepath))
```

Comment:
This error message doesn't format wonderfully - should look to word it differently.